ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryCleanupService.java

```diff
@@ -114,32 +122,35 @@ public String getAppId() {
     @Override
     public void run() {
       try {
-        final Path pathName = Paths.get(stateDir + "/" + appId);
+        final Path pathName = Paths.get(this.pathName);
         final File directory = new File(String.valueOf(pathName.normalize()));
         if (directory.exists()) {
           FileUtils.deleteDirectory(directory);
           LOG.warn("Deleted local state store for non-existing query {}. "
                   + "This is not expected and was likely due to a "
                   + "race condition when the query was dropped before.",
-              appId);
+              queryTopicPrefix);
         }
       } catch (Exception e) {
-        LOG.error("Error cleaning up state directory {}\n. {}", appId, e);
+        LOG.error("Error cleaning up state directory {}\n. {}", pathName, e);
       }
       tryRun(
           () -> SchemaRegistryUtil.cleanupInternalTopicSchemas(
-            appId,
+            queryTopicPrefix,
             serviceContext.getSchemaRegistryClient(),
             isTransient),
           "internal topic schemas"
       );
 
-      tryRun(() -> serviceContext.getTopicClient().deleteInternalTopics(appId), "internal topics");
-      tryRun(
-          () -> serviceContext
-              .getConsumerGroupClient()
-              .deleteConsumerGroups(ImmutableSet.of(appId)),
-          "internal consumer groups");
+      tryRun(() -> serviceContext.getTopicClient().deleteInternalTopics(queryTopicPrefix),
+          "internal topics");
+      if (!topologyName.isPresent()) {
```

Comment:
Are we not cleaning up consumer groups because there might be shared runtimes that are still using them? Is there a scenario where there's only one runtime left so we would want to clean up consumer groups?

Reply:
yes. That is not really a seniario, the number of runtimes should be fixed and we should not clean them when we are sharing them